src/main/java/util/ApiQuotaManager.java

```diff
@@ -0,0 +1,169 @@
+        this.refreshTimer = new TickingTimer("Refresh Timer",
+                (int) Utility.minsToSecs(DEFAULT_REFRESH_PERIOD_IN_MINS),
+                UI.status::updateTimeToRefresh, logic::refresh, TimeUnit.SECONDS);
+        this.refreshTimer.start();
+        UI.events.registerEvent((RefreshTimerTriggeredEventHandler) this::updateSyncRefreshRate);
+    }
+
+    /**
+     * @return The duration between each refresh of the data store.
+     */
+    public long getRefreshDurationInMinutes() {
+        return refreshDurationInMinutes;
+    }
+
+    /**
+     * Restarts the refreshTimer.
+     */
+    public void restartRefreshTimer(){
+        this.refreshTimer.restart();
+    }
+
+    /**
+     * Updates the period of the refreshTimer for synchronization of the data store.
+     * @param e RefreshTimerTriggeredEvent object that holds the current API quota information.
+     */
+    private void updateSyncRefreshRate(RefreshTimerTriggeredEvent e) {
+        ApiQuotaInfo info = e.getApiQuotaInfo();
+        apiCallsUsedInPreviousRefresh = computeApiCallsUsedInPreviousRefresh(info.getRemainingRequests());
+        refreshDurationInMinutes = computeRefreshTimerPeriod(info.getRemainingRequests(),
+                Utility.minutesFromNow(info.getNextRefreshInMillisecs()),
+                apiCallsUsedInPreviousRefresh,
+                API_QUOTA_BUFFER,
+                DEFAULT_REFRESH_PERIOD_IN_MINS);
+        this.refreshTimer.restartTimer((int) Utility.minsToSecs(refreshDurationInMinutes));
+        logger.info("Refresh period updated to " + refreshDurationInMinutes
+                + "mins with API calls used in previous refresh cycle is " + apiCallsUsedInPreviousRefresh
+                + ", current API quota is " + info.getRemainingRequests() + " and next API quota top-up in "
+                + info.getNextRefreshInMinutesFromNow() + "mins.");
+    }
+
+    /**
+     * Computes the API calls used in previous refresh.
+     * @param remainingRequests The number of API requests remaining in the current API quota window.
+     * @return The number of API calls used in previous refresh.
+     */
+    private int computeApiCallsUsedInPreviousRefresh(int remainingRequests) {
+        int difference = previousRemainingApiRequests - remainingRequests;
+        previousRemainingApiRequests = remainingRequests;
+
+        if (difference >= 0) {
+            return difference;
+        }
+        return apiCallsUsedInPreviousRefresh;
+    }
+
+    /**
+     * Computes the TickerTimer period that is used for refreshing the issues periodically
+     * Assumes future refreshes will take the same number of API calls as the last refresh and find out the refresh
+     * duration that will spread out the refreshes until the next API quota top up.
+     *
+     * For some cases where the computed refresh rate is equal to the remainingTimeInMins,
+     * bufferTime is added to ensure that the next refresh happens after the apiQuota renewal.
```

Comment:
1. Right, changed it
2. It just for efficiency purpose. If refresh happens before the api topup, it will most likely use up all the reserved api quota for manual operation and schedule the refresh again based on the new parameter.(most likely 1 min later)